### ISSUE TITLE
[VITE] Libraries shipping vanilla extract css loaded out of order when using vite

### ISSUE THREAD

**Issue by otaviosoares** [vanilla-extract-css/vanilla-extract#1376]
### Describe the bug

It happens when using vite and importing VE css from a library. The css is loaded in a non deterministic way, causing reset css to have a higher priority.

I created a small example using braid-design-system. Apparently, vite's dependency optimizer is bundling the styles in the wrong order, but I'm not sure.

![Screenshot 2024-04-05 at 10 59 25 AM](https://github.com/vanilla-extract-css/vanilla-extract/assets/180912/c5e1e339-d46f-4f9b-bd4b-b6cd68a3759e)

![Screenshot 2024-04-05 at 10 59 37 AM](https://github.com/vanilla-extract-css/vanilla-extract/assets/180912/07cce585-77b1-402a-8c98-a4318c4133e9)

**DISCLAIMER**: This only happens in dev mode. During build the css file that is extract has the correct order.

**WORKAROUND** I'm using css layers to avoid this issue but I'm intrigued and would love to know what's causing this behaviour. Any help is appreciated.

### Reproduction

https://stackblitz.com/edit/vitejs-vite-qfbgzz

### System Info

```shell
System:
    OS: macOS 14.2.1
    CPU: (16) arm64 Apple M3 Max
    Memory: 3.28 GB / 128.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    Yarn: 4.0.2 - ~/.yarn/bin/yarn
    npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
    bun: 1.0.0 - ~/.bun/bin/bun
  Browsers:
    Chrome: 123.0.6312.107
    Edge: 123.0.2420.81
    Safari: 17.2.1
  npmPackages:
    vite: ^5.1.4 => 5.1.4
```

### Used Package Manager

yarn

### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by rururux** [comment]
maybe a duplicate of #725

**Comment by askoufis** [comment]
You're correct that vite's dependency optimization is to blame here. This is our current workaround for this issue.

```ts
import { createRequire } from 'node:module';
import { dirname } from 'node:path';
import { cssFileFilter } from '@vanilla-extract/integration';

const require = createRequire(import.meta.url);

const fixViteVanillaExtractDepScanPlugin = (): Plugin => ({
  name: 'fix-vite-vanilla-extract-dep-scan',
  setup(build) {
    build.onResolve(
      { filter: cssFileFilter },
      ({ importer, path: pathName }) => ({
        path: require.resolve(pathName, { paths: [dirname(importer)] }),
        external: true,
      })
    );
  },
});
```
